_tests_/ship.test.js

```diff
@@ -0,0 +1,126 @@
+const Ship = require('../src/ship');
+const Port = require('../src/port');
+const Itinerary = require('../src/itinerary');
+
+describe('constructor', () => {
+    let ship;
+    let port;
+    let dover;
+    let calais;
+    let itinerary;
+    beforeEach(() => {
+        port = {
+            removeShip: jest.fn(),
+            addShip: jest.fn(),
+        };
+        dover = {
+            ...port,
+            name: 'Dover',
+            ships: [],
+        }
+        calais = {
+            ...port,
+            name: 'Calais',
+            ships: [],
+        }
+
+        itinerary = {
+            ports: [dover, calais]
+        };
+    
+        // port = new Port('Dover');
```

Comment:
I see dead code 💀 

Commented out code should just be removed. We can always use git to revert back to a point in time if necessary.